packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -800,6 +814,30 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    /**
+     * Summarizer client should not have local changes. These changes can become part of the summary and can break
+     * eventual consistency. For example, the next summary (say at ref seq# 100) may contain these changes whereas
+     * other clients that are up-to-date till seq# 100 may not have them yet.
+     */
+    protected identifyLocalChangeInSummarizer(eventName: string, type?: string) {
+        if (this.clientDetails.type === summarizerClientType) {
+            // Log a telemetry if there are local changes in the summarizer. This will give us data on how often
+            // this is happening and which data stores do this. The eventual goal is to disallow local changes
+            // in the summarizer and the data will help us plan this.
+            this.mc.logger.sendTelemetryEvent({
```

Comment:
do you want your other PR here that limits number of these events?

Reply:
yes, I ported those changes as well.